### DISCUSSION TITLE
Moving a CherryTree database to Trilium ?

### DISCUSSION THREAD

**Discussion by ghost** [TriliumNext/Trilium#4038] [Q&A] [answered]
Hi all, 

Has any of you experienced moving a CherryTree database to Trilium ? 
I would really appreciate if you could share your method. I'd like to avoid recreating the whole tree structure into Trilium, and importing html files one by one... (the only way I've found so far). 

@tdlf32, you mentioned in issue #3682 :
> For the moment I'm using CherryTree, but as soon as trilium has this update, I'll migrate my entire cherrytree database to trilium!

Have you tried yet ?

Some ideas I've been thinking of : 
- create the entire note tree manually in Trilium, and import HTML files one by one (tedious)
- recreate an SQL database in Trilium format, from the CherryTree SQLite database. Any hint on how to do this ? 
- import the HTML files into another tool that can read it, alongside the tree structure (Evernote ? Joplin ?), and then export it as a file that Trilium can import

The export options of CherryTree are : 
- PDF
- HTML with .css and folder with embeded images and files
  - all notes in separate files (including an index from which we could recreate the tree)
  - HTML with a unique file (tree structure is lost), 
- SQLite db file, with CherryTree database structure
- text format (separate files or unique file)

Any clue would be appreciated.

Thanks !

**Comment by meichthys** [comment]
Trilium can import markdown, so this may get you a step closer: https://gitlab.com/kibley/cherrytreetomarkdown/-/tree/master/examples/oscp-lab-report

**Comment by Nriver** [comment]
See https://github.com/zadam/trilium/issues/3617#issuecomment-1432673307

**Comment by ghost** [comment]
Thanks ! I'll try to go this way.

**Comment by obuno** [comment]
Hi all,

I've tried the cherrytree to markdown possibility mentioned above but wasn't satisfied with its results, in my testing it needed a lot of manual interactions (test from cherrytree to obsidian) post migration. Finally I went for cherrytree to TXT exports skipping the images (haven't got loads TBH) -- zipping the exports in a single .zip and imported that within Trilium -- I'm then re-editing/classifying to satisfaction on each search results that submerge up older cherrytree based items.

My cherrytree DB is about roughly 50MiB filled mainly with technical info's/text.

My main reason for migrating was random cherrytree DB corruptions which ended up to leave me sour... It's been hard for me to move away from cheerytree (in favor of a decent replacement) though Trilium is what I've been after and I'm very satisfied.

**Comment by ShellUnease** [comment, marked as answer]
https://github.com/ShellUnease/cherrytree2trilium - I made adjustments to an old cherrytree2markdown project and added some features to migrate my 363MB CherryTree db to Trilium while maintaining formatting, tree structure, images and even internal link nodes. It may or may not be rough around the edges but maybe it helps someone else to migrate too :pray:

**Reply by eliandoran** [reply]
@ShellUnease , sounds great. :)
Hope you enjoy Trilium!
